matrix/plugins/actions.py

```diff
@@ -0,0 +1,45 @@
+from datetime import timedelta, datetime
+
+from .glitch import glitch  # noqa
+
+
+async def deploy(context, rule, action, event=None):
+    rule.log.info("Deploying %s", action.args['entity'])
+    context.apps.extend(await context.juju_model.deploy(action.args['entity']))
+    rule.log.info("Deploy COMPLETE")
+    return True
+
+
+async def health(context, rule, action, event=None):
+    if not context.apps:
+        # this shouldn't happen, because the health rule has
+        # an "after: deploy", but that doesn't seem to be honored
+        return
+    stable_period = timedelta(seconds=30)
+    errored_units = []
+    busy_units = []
+    for app in context.apps:
+        for unit in app.units:
+            now = datetime.utcnow()
+            agent_status_duration = now - unit.agent_status_since
```

Comment:
Will we hit stable state if glitch is running? The agents might be perturbed a good deal, even on minor actions. I suppose we should if its working as intended